src/main/java/jenkins/bouncycastle/api/BCPasswordException.java

```diff
@@ -24,6 +24,9 @@
 
 package jenkins.bouncycastle.api;
 
+/**
+ * Indicates a password or passphrase is needed and none or a wrong one was provided
+ */
 public class BCPasswordException extends RuntimeException {
```

Comment:
`java.security.UnrecoverableEntryException` or `java.security.UnrecoverableKeyException` is an existing exception that matches exactly the use case you had for this.